configure.ac

```diff
@@ -36,7 +36,7 @@ AX_CUDA
 if test "$want_cuda" = "yes" ; then
   CPP_STANDARD=17
 else
-  CPP_STANDARD=11
+  CPP_STANDARD=17
```

Comment:
We need to make sure this will work with whatever compiler @timothygrant80 plans to use in the virtual machine approach to compiling for distribution. If it is okay, then we should just get rid of the logic in 36 and set the value

Reply:
Ok, how about we change to logic for now to `want_cuda or cxx=clang++`?